packages/components/src/templates/next/components/complex/CollectionBlock/CollectionBlock.tsx

```diff
@@ -0,0 +1,221 @@
+  shouldLazyLoad,
+  numberOfCards,
+}: CollectionBlockSingleCardProps): JSX.Element => {
+  const isExternalLink = !!referenceLinkHref && isExternalUrl(referenceLinkHref)
+
+  const renderImage = () => {
+    const hasImage = image?.src !== undefined
+    const imageSrc = hasImage ? image.src : site.logoUrl
+    const imageAlt = hasImage ? image.alt : `Site logo for ${site.siteName}`
+
+    return (
+      <div className={compoundStyles.cardImageContainer({ numberOfCards })}>
+        <ImageClient
+          src={
+            isExternalUrl(imageSrc) || site.assetsBaseUrl === undefined
+              ? imageSrc
+              : `${site.assetsBaseUrl}${imageSrc}`
+          }
+          alt={imageAlt}
+          width="100%"
+          className={compoundStyles.cardImage({
+            imageFit: hasImage ? "cover" : "contain",
+          })}
+          lazyLoading={shouldLazyLoad}
+          assetsBaseUrl={site.assetsBaseUrl}
+        />
+      </div>
+    )
+  }
+
+  return (
+    <Link
+      href={referenceLinkHref}
+      className={compoundStyles.cardContainer()}
+      LinkComponent={LinkComponent}
+      isExternal={isExternalLink}
+    >
+      {displayThumbnail && renderImage()}
+      <div className={compoundStyles.cardTextContainer()}>
+        {lastUpdated && (
+          <p className={compoundStyles.cardLastUpdated()}>
+            {getFormattedDate(lastUpdated)}
+          </p>
+        )}
+
+        <h3 className={compoundStyles.cardTitle()}>
+          {title}
+          <BiRightArrowAlt
+            aria-hidden
+            className={compoundStyles.cardTitleArrow({
+              isExternalLink,
+            })}
+          />
+        </h3>
+
+        {displayCategory && category && (
+          <p className={compoundStyles.cardCategory()}>{category}</p>
+        )}
+      </div>
+    </Link>
+  )
+}
+
+export const CollectionBlock = ({
+  site,
+  LinkComponent,
+  collectionReferenceLink,
+  customTitle,
+  customDescription,
+  displayThumbnail,
+  displayCategory,
+  buttonLabel,
+  shouldLazyLoad,
+}: CollectionBlockProps): JSX.Element => {
+  const collectionId = getResourceIdFromReferenceLink(collectionReferenceLink)
+
+  const collectionParent = getCollectionParent({ site, collectionId })
+
+  const collectionPages = getCollectionPages({
+    site,
+    collectionParent,
+  })
+
+  if (collectionPages.length === 0) {
+    return <></>
+  }
+
+  const numberOfCards =
+    collectionPages.length as CollectionBlockNumberOfCards["numberOfCards"]
+
+  return (
+    <section className={compoundStyles.container()}>
+      <div className={compoundStyles.headingContainer()}>
+        <h2 className={compoundStyles.headingTitle()}>
+          {customTitle ?? collectionParent.title}
+        </h2>
+        <p>{customDescription ?? collectionParent.summary}</p>
+      </div>
+
+      <div className={compoundStyles.grid({ numberOfCards })}>
+        {collectionPages.map((card, idx) => (
+          <SingleCard
+            key={idx}
```

Comment:
issue: I think we should use `card.id` instead of the index, as index is not recommended to be a key in React

Reply:
updated in https://github.com/opengovsg/isomer/pull/1250/commits/406048d8b434aaeeae82b6c55360835e6f9fc009